src/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -337,109 +332,183 @@ contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistr
-        }));
-
-        // if the operator is not already registered, then they are registering for the first time
-        if (_operatorInfo[operator].status != OperatorStatus.REGISTERED) {
-            _operatorInfo[operator] = Operator({
-                operatorId: operatorId,
+        if (_operatorInfo[operatorToRegister].status != OperatorStatus.REGISTERED) {
+            _operatorInfo[operatorToRegister] = Operator({
+                operatorId: idToRegister,
                 status: OperatorStatus.REGISTERED
             });
 
-            emit OperatorRegistered(operator, operatorId);
+            emit OperatorRegistered(operatorToRegister, idToRegister);
+        }
+
+        /**
+         * Register the operator with the BLSPubkeyRegistry, StakeRegistry, and IndexRegistry
+         */
+        bytes32 registeredId = blsPubkeyRegistry.registerOperator(operatorToRegister, quorumNumbers);
+        require(registeredId == idToRegister, "BLSRegistryCoordinatorWithIndices._registerOperator: operatorId mismatch");
+        (uint96[] memory registeredStakes, uint96[] memory totalStakes) = 
+            stakeRegistry.registerOperator(operatorToRegister, idToRegister, quorumNumbers);
+        uint32[] memory numOperatorsPerQuorum = indexRegistry.registerOperator(idToRegister, quorumNumbers);
+
+        /**
+         * Now that we've registered the operator, validate that the new total operators
+         * in each quorum do not exceed the maximum. 
+         * 
+         * If they do and we're performing churn, we deregister the corresponding operator 
+         * specified in `operatorKickParams`
+         */
+
+        uint256 kickIndex = 0;
+        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            
+            OperatorSetParam memory operatorSetParams = _quorumParams[quorumNumber];
+            bool exceedsOperatorCapacity = numOperatorsPerQuorum[i] > operatorSetParams.maxOperatorCount;
+
+            /**
+             * If we exceed the max operator count for this quorum, we need to replace an existing operator
+             * with a new one.
+             */
+            if (exceedsOperatorCapacity) {
+                require(performChurn, "BLSRegistryCoordinatorWithIndices._registerOperator: churn required for overfilled quorum");
+                
+                // Validate that `operatorToRegister` can replace the operator specified in the kick params
+                _validateChurn({
+                    quorumNumber: quorumNumber,
+                    totalQuorumStake: totalStakes[i],
+                    newOperator: operatorToRegister,
+                    newOperatorStake: registeredStakes[i],
+                    kickParams: operatorKickParams[i],
+                    setParams: operatorSetParams
+                });
+
+                _deregisterOperator(operatorKickParams[i].operator, quorumNumbers[i:i+1]);
```

Comment:
is `quorumNumbers[i:i+1]` the correct syntax, or should it just be `quorumNumbers[i]`?
I feel like this might be doing 2 quorums instead of 1 right now?

Reply:
actually just borrowed this from the current implementation. it looks like the correct syntax to get a slice of length 1, but i'd hope we're testing it somewhere :D